### DISCUSSION TITLE
OBJ Support for Manim

### DISCUSSION THREAD

**Discussion by MrDiver** [ManimCommunity/manim#2835] [Suggestions and Proposals]
- #2030

> ## Overview: What does this pull request change?
> 
> Add an `OBJMobject` which allows users to make use of 3D models from `.obj` files.
> ## Motivation and Explanation: Why and how do your changes improve the library?
> 
> It's tedious and laborious to make complex 3D models in Manim by hand. One can now make a 3D model in 3rd party software better suited to the task (such as Blender) and import the model into Manim using `OBJMobject`.
> 
> One can use `OBJMobject` like so:
> 
> ```python
> class Test(ThreeDScene):
>     def construct(self):
>         self.set_camera_orientation(
>             phi=-45 * DEGREES,
>             theta=-135 * DEGREES,
>             gamma=-55 * DEGREES)
>         model = OBJVMobject("/Users/aathishs/Projects/Python/ManimStuff/manim-projects/threedfilemob/airboat.obj").scale(0.5)
>         self.play(Write(model))
>         self.play(Rotate(model,axis=UP,angle=2*PI),run_time=5,rate_func=linear)
>         self.wait(1)
> ```
> 
> The Output for the code above is:
> Test.mp4
> ## Links to added or changed documentation pages
> 
> Documentation is WIP
> ## Further Information and Comments
> 
> The `.obj` file can be found at :
> 
> https://people.sc.fsu.edu/~jburkardt/data/obj/airboat.obj
> 
> And the `.mtl` file for textures is available at:
> 
> https://people.sc.fsu.edu/~jburkardt/data/obj/vp.mtl
> 
> NOTES: The colour `bronze` has to be changed to `brass` in the `.obj` file for colours to render properly. The `vp.mtl` file must be renamed to `airboat.mtl` and be placed next to the `airboat.obj` file for colours to render properly.
> 
> This is very much a work in progress. OpenGL support is untested, and much of the OBJ file spec is not yet implemented. The OBJ file specifications can be found here: http://paulbourke.net/dataformats/obj/ In the near future, I'll add a checklist of stuff that's yet to be implemented from the `.obj` and `.mtl` file spec.
> ## Reviewer Checklist
> 
>     * [ ]  The PR title is descriptive enough for the changelog, and the PR is labeled correctly
> 
>     * [ ]  If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
> 
>     * [ ]  If applicable: newly added functions and classes are tested

**Comment by hisacro** [comment]
any updates?
